llvm/lib/CodeGen/GlobalISel/InlineAsmLowering.cpp

```diff
@@ -207,7 +142,16 @@ static void computeConstraintToUse(const TargetLowering *TLI,
     OpInfo.ConstraintCode = OpInfo.Codes[0];
     OpInfo.ConstraintType = TLI->getConstraintType(OpInfo.ConstraintCode);
   } else {
-    chooseConstraint(OpInfo, TLI);
+    TargetLowering::ConstraintGroup G = TLI->getConstraintPreferences(OpInfo);
+    if (G.empty())
+      return;
+    // FIXME: prefer immediate constraints if the target allows it
+    unsigned BestIdx = 0;
+    while (G[BestIdx].second == TargetLowering::C_Other ||
+           G[BestIdx].second == TargetLowering::C_Immediate)
```

Comment:
I'm nervous when I see loops without clear bound.
Why don't we have to check that `BestIdx < G.size()` to prevent out-of-bounds accesses?

Reply:
https://github.com/llvm/llvm-project/pull/67057/commits/a0cb4cb60f3ee83876929a152ed0a11abe34fe6c